src/Compilers/CSharp/Portable/Syntax/SyntaxNormalizer.cs

```diff
@@ -485,9 +491,108 @@ private static bool NeedsSeparator(SyntaxToken token, SyntaxToken next)
+                switch (token.Kind())
+                {
+                    // No spacing between the * and < tokens if there is no calling convention
+                    case SyntaxKind.AsteriskToken:
+                    // No spacing between the calling convention and opening angle bracket of function pointer types:
+                    // delegate* managed<
+                    case SyntaxKind.ManagedKeyword:
+                    case SyntaxKind.UnmanagedKeyword:
+                    // No spacing between the calling convention specifier and the opening angle
+                    // delegate* unmanaged[Cdecl]<
+                    case SyntaxKind.CloseBracketToken when token.Parent.IsKind(SyntaxKind.FunctionPointerUnmanagedCallingConventionList):
+                        return false;
+                }
+            }
+
+            // No space between unmanaged and the [
+            // delegate* unmanaged[
+            if (token.Parent.IsKind(SyntaxKind.FunctionPointerCallingConvention) && next.Parent.IsKind(SyntaxKind.FunctionPointerUnmanagedCallingConventionList) &&
+                next.IsKind(SyntaxKind.OpenBracketToken))
+            {
+                return false;
+            }
+
+            // Function pointer calling convention adjustments
+            if (next.Parent.IsKind(SyntaxKind.FunctionPointerUnmanagedCallingConventionList) && token.Parent.IsKind(SyntaxKind.FunctionPointerUnmanagedCallingConventionList))
+            {
+                if (next.IsKind(SyntaxKind.IdentifierToken))
+                {
+                    if (token.IsKind(SyntaxKind.OpenBracketToken))
+                    {
+                        return false;
+                    }
+                    // Space after the ,
+                    // unmanaged[Cdecl, Thiscall
+                    else if (token.IsKind(SyntaxKind.CommaToken))
+                    {
+                        return true;
+                    }
+                }
+
+                // No space between identifier and comma
+                // unmanaged[Cdecl,
+                if (next.IsKind(SyntaxKind.CommaToken))
+                {
+                    return false;
+                }
+
+                // No space before the ]
+                // unmanaged[Cdecl]
+                if (next.IsKind(SyntaxKind.CloseBracketToken))
+                {
+                    return false;
+                }
+            }
+
+            // No space after the < in function pointer parameter lists
+            // delegate*<void
+            if (token.IsKind(SyntaxKind.LessThanToken) && token.Parent.IsKind(SyntaxKind.FunctionPointerParameterList))
+            {
+                return false;
+            }
+
+            // No space before the > in function pointer parameter lists
+            // delegate*<void>
+            if (next.IsKind(SyntaxKind.GreaterThanToken) && next.Parent.IsKind(SyntaxKind.FunctionPointerParameterList))
+            {
+                return false;
```

Comment:
The IDE spacing is using an IDE option, which I don't think is available to use here. I defaulted to the more common convention, which is no space before `>` and no space after `<`.

Reply:
Yeah, the normalizer isn't customized. There's one normalization.

---
In reply to: [598105046](https://github.com/dotnet/roslyn/pull/51981#discussion_r598105046) [](ancestors = 598105046)